### PR TITLE
Location services

### PR DESCRIPTION
**Pull request by TurtleSquad007**

### CODE REVIEW

**Comment by TurtleSquad007** [comment]
TODO: Add edit location dialog 
TODO: Add better warning display -> Might leave as is; not sure how much time we have.

File: code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java [hunk 1]
```diff
@@ -187,6 +189,26 @@ private void setButtons() {
...
+            navController.navigate(R.id.map_display_fragment, locArgs);
+            parentActivity.setCurrentScreen(Screen.MAP);
+            parentActivity.setCurrentFragment(this);
+            parentActivity.findViewById(R.id.fab_button).setVisibility(View.GONE);
+        });
+
+        if (experimentStringId != null) {
+            update(experimentStringId);
+        }
+        return root;
```
Comment (Pegmode) [inline code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java:211]: This line is giving me this error 
`Experiment-Automata\code\app\src\main\java\com\example\experiment_automata\ui\experiments\NavExperimentDetailsFragment.java:211: error: incompatible types: unexpected return value
        return root;`

Comment (TurtleSquad007) [reply code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java:211]: Can you define the steps you took to get that error?

Comment (Pegmode) [reply code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java:211]: I ran the app in emulator and immediately it displays this error

Comment (TurtleSquad007) [reply code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java:211]: Oh that's interesting. Justen didn't get that.

Comment (TurtleSquad007) [reply code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java:211]: Are you sure you're running the latest pull?

Comment (TurtleSquad007) [reply code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java:211]: Oh lol. It was me who had the bad pull. Let me push the fix

Comment (Pegmode) [reply code/app/src/main/java/com/example/experiment_automata/ui/experiments/NavExperimentDetailsFragment.java]: Runs fine now

**Review by Pegmode — APPROVED** [review]
Works fine now. Tested all trial types and they work correctly with location services

**Review by NickNissen — APPROVED** [review]
error appears to be fixed, location services works on my testing

**Pull request merged**